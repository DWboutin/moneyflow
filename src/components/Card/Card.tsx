import React, { ReactNode } from "react";
import styled from "styled-components";
import { ThemeContainer } from "../../styles/styles";

interface ContainerProps {}

const Container = styled.div<ContainerProps>`
  display: flex;
  padding: 40px;
  border-radius: 0px 40px 40px 40px;
  background-color: ${({ theme }: ThemeContainer) => theme.card.bg};
  box-shadow: ${({ theme }: ThemeContainer) => theme.card.boxShadow};
`;

export interface Props {
  children: ReactNode;
}

const Card: React.FunctionComponent<Props> = ({ children }) => {
  return <Container>{children}</Container>;
};

export default Card;
