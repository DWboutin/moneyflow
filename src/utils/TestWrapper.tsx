import React, { ReactNode } from "react";
import { Provider } from "react-redux";
import { ToastContainer } from "react-toastify";
import { ThemeProvider } from "styled-components";

import store from "../store/store";
import GlobalStyle from "../styles/globalStyle";

import { theme } from "../styles/styles";

interface Props {
  children: ReactNode;
}

const TestWrapper: React.FunctionComponent<Props> = ({ children }) => (
  <Provider store={store}>
    <ThemeProvider theme={theme}>
      <GlobalStyle />
      {children}
      <ToastContainer />
    </ThemeProvider>
  </Provider>
);

export default TestWrapper;
