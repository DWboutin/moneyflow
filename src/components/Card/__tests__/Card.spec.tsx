import React from "react";
import { render, RenderResult } from "@testing-library/react";

import Card, { Props } from "../Card";
import TestWrapper from "../../../utils/TestWrapper";

describe("<Card />", () => {
  let renderResult: RenderResult;

  const renderCard = ({ children }: Props) => {
    renderResult = render(
      <TestWrapper>
        <Card>{children}</Card>
      </TestWrapper>
    );
  };

  describe("default rendering", () => {
    beforeEach(() => {
      renderCard({ children: "Content" });
    });

    it("should match snapshot", () => {
      expect(renderResult.container).toMatchSnapshot();
    });
  });
});
