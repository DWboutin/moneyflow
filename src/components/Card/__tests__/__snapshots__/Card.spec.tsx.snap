// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> default rendering should match snapshot 1`] = `
<div>
  <div
    class="sc-bdvvtL gUCkrw"
  >
    Content
  </div>
  <div
    class="Toastify"
  />
</div>
`;
